### PR TITLE
MM-32162 Review and Update mmctl product docs

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation task: https://mattermost.atlassian.net/browse/MM-32162

Updated:
- Reviewed all mmctl commands against the 'mmctl' repository and updated where needed.

### CODE REVIEW

File: source/administration/mmctl-cli-tool.rst [hunk 1]
```diff
@@ -3980,35 +4147,154 @@ mmctl user activate
...
 
 **Examples**
 
 .. code-block:: sh
 
-    user create --email user@example.com --username userexample --password Password1
+   # You can create a user
+   $ mmctl user create --email user@example.com --username userexample --password Password1
+
+   # You can define optional fields like first name, last name and nick name too
```
Comment (justinegeffen) [inline source/administration/mmctl-cli-tool.rst:4283]:
```suggestion
   # You can define optional fields like first name, last name, and nickname
```

File: source/administration/mmctl-cli-tool.rst [hunk 2]
```diff
@@ -3980,35 +4147,154 @@ mmctl user activate
...
+   $ mmctl user create --email user@example.com --username userexample --password Password1 --firstname User --lastname        Example --nickname userex
+
+   # Also you can create the user as system administrator
```
Comment (justinegeffen) [inline source/administration/mmctl-cli-tool.rst:4286]:
```suggestion
   # You can also create the user as a System Admin
```

File: source/administration/mmctl-cli-tool.rst [hunk 3]
```diff
@@ -3980,35 +4147,154 @@ mmctl user activate
...
+   $ mmctl user create --email user@example.com --username userexample --password Password1 --system-admin
+
+   # Finally you can verify user on creation if you have enough permissions
```
Comment (justinegeffen) [inline source/administration/mmctl-cli-tool.rst:4289]:
```suggestion
   # Finally you can verify user on creation if you have the correct permissions
```

File: source/administration/mmctl-cli-tool.rst [hunk 4]
```diff
@@ -3980,35 +4147,154 @@ mmctl user activate
...
+   $ mmctl user create --email user@example.com --username userexample --password Password1 --system-admin --email-verified
 
 **Options**
 
 .. code-block:: sh
 
    --email string       Required. The email address for the new user account
+   --email_verified     Optional. If supplied, the new user will have the email verified. Defaults to false
```
Comment (justinegeffen) [inline source/administration/mmctl-cli-tool.rst:4297]:
```suggestion
   --email_verified     Optional. If supplied, the new user will have the email verified. Defaults to "false"
```

**Review by justinegeffen — COMMENTED** [review]
Thanks for this, @cwarnermm! It was a big PR. Apologies for the delayed review. Let me know if any of the suggestions I've made don't work. :)

**Comment by cwarnermm** [comment]
Thank you so much, @justinegeffen, for reviewing this huge PR for the `mmctl` documentation.

**Review by adamjclarkson — APPROVED** [review]

**Review by adamjclarkson — APPROVED** [review]

**Review by mgdelacroix — APPROVED** [review]
LGTM! :+1: Awesome work!

**Comment by mgdelacroix** [comment]
@cwarnermm I've ported back to the `mmctl` autogenerated docs some of the changes of this PR on https://github.com/mattermost/mmctl/pull/348

File: source/administration/mmctl-cli-tool.rst [hunk 5]
```diff
@@ -4735,160 +4982,205 @@ mmctl user deleteall
...
+  -h, --help   help for demote
 
 **Options inherited from parent commands**
 
+   --config-path string           path to the configuration directory. If "$HOME/.mmctl" exists it will take precedence over the default value (default "$XDG_CONFIG_HOME")
+   --format string                the format of the command output [plain, json] (default "plain")
+   --insecure-sha1-intermediate   allows to use insecure TLS protocols, such as SHA-1
+   --insecure-tls-version         allows to use TLS versions 1.0 and 1.1
+   --local                        allows communicating with the server through a unix socket
+   --strict                       will only run commands if the mmctl version matches the server one
```
Comment (amyblais) [inline source/administration/mmctl-cli-tool.rst:5031]: @cwarnermm This section looks broken https://docs.mattermost.com/administration/mmctl-cli-tool.html#mmctl-user-demote

Comment (cwarnermm) [reply source/administration/mmctl-cli-tool.rst:5031]: Thanks, @amyblais. Fixing via https://github.com/mattermost/docs/pull/4485

**Pull request merged**